src/core/server/models/story/story.ts

```diff
@@ -727,3 +735,124 @@ export async function retrieveStorySections(
   // the null entries, the resulting array can not contain null.
   return results.filter((section) => section !== null).sort() as string[];
 }
+
+export async function markStoryForArchiving(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+      isArchiving: { $in: [null, false] },
+      isArchived: { $in: [null, false] },
+    },
+    {
+      $set: {
+        isArchiving: true,
+        closedAt: now,
+        updatedAt: now,
+        startedArchivingAt: now,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  return result.value;
+}
+
+export async function markStoryForUnarchiving(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+      isArchiving: false,
+      isArchived: true,
+    },
+    {
+      $set: {
+        isArchiving: true,
+        closedAt: now,
+        updatedAt: now,
+        startedUnarchivingAt: now,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  return result.value;
+}
+
+export async function retrieveStoriesToBeArchived(
+  mongo: Db,
+  tenantID: string,
+  olderThan: Date
+): Promise<Cursor<Readonly<Story>>> {
+  const stories = collection(mongo).find({
+    tenantID,
+    createdAt: { $lte: olderThan },
+    isArchiving: { $in: [null, false] },
+    isArchived: { $in: [null, false] },
+    startedUnarchivingAt: { $in: [null, false] },
+    unarchivedAt: { $in: [null, false] },
+  });
+
+  return stories;
+}
+
+export async function markStoryAsArchived(
+  mongo: Db,
+  tenantID: string,
+  storyID: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    { id: storyID, tenantID },
+    {
+      $set: {
+        isArchiving: false,
+        isArchived: true,
+        archivedAt: now,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  return result.value;
+}
+
+export async function markStoryAsUnarchived(
+  mongo: Db,
+  tenantID: string,
+  storyID: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    { id: storyID, tenantID },
+    {
+      $set: {
+        isArchiving: false,
+        isArchived: false,
+        unarchivedAt: now,
```

Comment:
I'm not sure I like removing this metadata. It would prevent me from seeing when a story was archived and when it is un-archived. @wyattjoh largely asked me to add these in to aid in recover-ability and debugging should archiving issues crop up. Removing this data doesn't make the document more confusing in my mind. The reason being that if we have an `archivedAt` date that is less than the `unarchivedAt` date, we have all the data we need to grab both the archiving and un-archiving logs from Sentry/Cloud-Console. This seems like a good thing to keep.